CMakeLists.txt

```diff
@@ -1,131 +1,192 @@
-find_package(Pangolin REQUIRED)
-find_package(realsense2)
+#target_include_directories(${PROJECT_NAME} PRIVATE ${EIGEN3_INCLUDE_DIR})
+message(STATUS "EIGEN3_INCLUDE_DIR: ${EIGEN3_INCLUDE_DIR}")
+message(STATUS "Eigen3_VERSION: ${Eigen3_VERSION}")
+target_link_libraries(${PROJECT_NAME} PUBLIC Eigen3::Eigen)
 
-include_directories(
-${PROJECT_SOURCE_DIR}
-${PROJECT_SOURCE_DIR}/include
-${PROJECT_SOURCE_DIR}/include/CameraModels
-${EIGEN3_INCLUDE_DIR}
-${Pangolin_INCLUDE_DIRS}
-)
+find_package(Pangolin REQUIRED)
+#target_include_directories(${PROJECT_NAME} PRIVATE ${Pangolin_INCLUDE_DIRS})
+message(STATUS "Pangolin_LIBRARIES: ${Pangolin_LIBRARIES}")
+message(STATUS "Pangolin_VERSION: ${Pangolin_VERSION}")
+target_link_libraries(${PROJECT_NAME} PUBLIC ${Pangolin_LIBRARIES})
 
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/lib)
+find_package(realsense2)
+if(realsense2_FOUND)
+  message(STATUS "realsense2_VERSION: ${realsense2_VERSION}")
+  target_link_libraries(${PROJECT_NAME} PRIVATE ${realsense2_LIBRARY})
+endif()
 
-add_library(${PROJECT_NAME} SHARED
-src/System.cc
-src/Tracking.cc
-src/LocalMapping.cc
-src/LoopClosing.cc
-src/ORBextractor.cc
-src/ORBmatcher.cc
-src/FrameDrawer.cc
-src/Converter.cc
-src/MapPoint.cc
-src/KeyFrame.cc
-src/Atlas.cc
-src/Map.cc
-src/MapDrawer.cc
-src/Optimizer.cc
-src/Frame.cc
-src/KeyFrameDatabase.cc
-src/Sim3Solver.cc
-src/Initializer.cc
-src/Viewer.cc
-src/ImuTypes.cc
-src/G2oTypes.cc
-src/CameraModels/Pinhole.cpp
-src/CameraModels/KannalaBrandt8.cpp
-src/OptimizableTypes.cpp
-src/MLPnPsolver.cpp
-src/TwoViewReconstruction.cc
-include/System.h
-include/Tracking.h
-include/LocalMapping.h
-include/LoopClosing.h
-include/ORBextractor.h
-include/ORBmatcher.h
-include/FrameDrawer.h
-include/Converter.h
-include/MapPoint.h
-include/KeyFrame.h
-include/Atlas.h
-include/Map.h
-include/MapDrawer.h
-include/Optimizer.h
-include/Frame.h
-include/KeyFrameDatabase.h
-include/Sim3Solver.h
-include/Initializer.h
-include/Viewer.h
-include/ImuTypes.h
-include/G2oTypes.h
-include/CameraModels/GeometricCamera.h
-include/CameraModels/Pinhole.h
-include/CameraModels/KannalaBrandt8.h
-include/OptimizableTypes.h
-include/MLPnPsolver.h
-include/TwoViewReconstruction.h
-include/Config.h
-)
+find_package(Boost REQUIRED COMPONENTS serialization)
+message(STATUS "Boost_LIBRARIES: ${Boost_LIBRARIES}")
+target_link_libraries(${PROJECT_NAME} PRIVATE ${Boost_LIBRARIES})
 
-add_subdirectory(Thirdparty/g2o)
+if(CMAKE_BUILD_TYPE MATCHES Debug)
+  find_package(Backward)
+  message(STATUS "BACKWARD_LIBRARIES: ${BACKWARD_LIBRARIES}")
+  target_link_libraries(${PROJECT_NAME} PUBLIC Backward::Backward)
+endif()
 
-target_link_libraries(${PROJECT_NAME}
-${OpenCV_LIBS}
-${EIGEN3_LIBS}
-${Pangolin_LIBRARIES}
-${PROJECT_SOURCE_DIR}/Thirdparty/DBoW2/lib/libDBoW2.so
-${PROJECT_SOURCE_DIR}/Thirdparty/g2o/lib/libg2o.so
--lboost_serialization
--lcrypto
-)
+# TODO: OpenSSL is used only in System.cc #include <openssl/md5.h>, but then MD5 is never used.
```

Comment:
FYI this is what I was worried about. Still playing around with it, but running out of ideas. Open to ideas...

Reply:
sorry just needed to screenshot something, ignore me :)